rfc/20241023-rfc-tracking-issues.md

```diff
@@ -0,0 +1,209 @@
+
+## Proposed Solution
+
+As noted above, this proposal effectively expands on what is currently the last step in the existing RFC process document:
+
+> 6. The RFC is Merged, and the Core Team creates issues in the relevant repositories to track the work required to implement the RFC.
+
+Under this proposal, immediately after merging an accepted RFC the core team creates a single "tracking issue" that represents the work to implement that RFC. The tracking RFC title is always the prefix "RFC tracker: " followed by the same title used in the RFC document itself. For now, tracking issues belong in the main `opentofu/opentofu` repository regardless of which codebases the RFC proposes to modify, though [we might change that in future](#moving-rfcs-and-their-tracking-issues-into-a-separate-repository).
+
+The tracking issue description includes the following information:
+
+- A link to the RFC text itself, and the PR(s) where the document was previously discussed.
+- Links to each of the feature request issues that were identified at the beginning of the RFC text, just to make backlinks appear on those other issues linking to the tracking issue.
+- A checkbox list of blockers that must be cleared before implementation could begin, or at least before any implementation could be merged. If the blockers are work described by other RFCs, these items link directly to each other RFC's tracking issue.
+
+    (It may be appropriate to begin a partial implementation that then remains unmerged until blockers are cleared, but the author must understand that their initial implementation may need significant revisions or replacement once the blockers have been resolved.)
+- A checkbox list summarizing the work that must be completed before the RFC is considered fully implemented. This includes both the direct development work and other related tasks such as updating the documentation and coordination with other projects that might be run concurrently.
+
+    In particular, any pull requests that are intended as progress towards the completion of the features described in the RFC should be directly linked as checklist items so that there is a single place to find work that was already completed and work that's awaiting review, to help avoid accidental duplicate or conflicting work between people working asynchronously.
+
+An RFC tracking issue is tagged with the same "rfc" label we currently use for RFC pull requests, since the tracking issue represents the remaining work related to a specific RFC. [A label search for "rfc"](https://github.com/opentofu/opentofu/labels/rfc) would therefore return both the not-yet-approved RFCs represented as pull requests and the accepted-but-not-completed RFCs represented as tracking issues.
+
+An RFC tracking issue should also have exactly one of the following labels summarizing its current status:
```

Comment:
The other projects I was taking inspiration from (mainly Rust and Go) use labels, though I must admit I don't know _why_ they use labels. I might guess that it's because it's the most prominent filtering mechanism for GitHub issues and so it's the most accessible to folks who aren't familiar with the fine details of a project's workflow.

With that said, I was intending to suggest that the labels be _in addition to_ a more structured project management tool like GitHub's Projects. My other section below about "roadmapping" talks a little about that, but I kept it light and unspecific for now: since our current representation of that is not public I expect that's something we'll want to evolve, but exactly how to evolve it seems like a much bigger discussion than how to represent the status of individual RFCs and so I'd prefer to leave that for a later RFC.